src/main/java/org/jenkinsci/test/acceptance/po/Jenkins.java

```diff
@@ -112,7 +113,14 @@ public void restart() {
         visit("restart");
         clickButton("Yes");
 
-        waitFor(driver, not(hasContent("Please wait")), JenkinsController.STARTUP_TIMEOUT);
+        try {
+            waitFor(driver, not(hasContent("Please wait")), JenkinsController.STARTUP_TIMEOUT);
+        }catch(TimeoutException e) {
+            //Let's try to avoid false negatives or not auto refresh
+            visit(driver.getCurrentUrl());
+            //we wait 10 seconds for refresh things.
+            waitFor(driver, hasContent("New Item"), 10);
```

Comment:
Yes the problem is as I explained in JIRA, a lot of test fails because a timeout error is thrown. Of course this sometimes happens and sometimes not, so since timeout is set to 360 seconds, probably the Jenkins is restarted but the screen has not been refreshed. So what I do in these cases is force a refresh.

Also notice that I am not changing anything as you pointed out, simply adding a manual refresh in case of error timeout error. So I am not modifying anything but trying to solve a problem. Of course maybe this is not the problem, but since we cannot reproduce since in local always work and in Jenkins sometimes yes sometimes no, the only thing to proceed to avoid this kind of errors taht comes to my mind is doing in this way.